public/app/features/explore/utils/ResultProcessor.ts

```diff
@@ -80,14 +80,19 @@ export class ResultProcessor {
     const graphInterval = this.state.queryIntervals.intervalMs;
     const dataFrame = this.rawData.map(result => guessFieldTypes(toDataFrame(result)));
     const newResults = this.rawData ? dataFrameToLogsModel(dataFrame, graphInterval) : null;
-    const sortedNewResults = sortLogsResult(newResults, this.state.refreshInterval);
+    const sortOrder = refreshIntervalToSortOrder(this.state.refreshInterval);
+    const sortedNewResults = sortLogsResult(newResults, sortOrder);
 
     if (this.replacePreviousResults) {
-      return sortedNewResults;
+      const slice = 1000;
```

Comment:
Found a bug, we didn't reduce the rows when replacing previous results which could potentially lead to lot's of rows

Reply:
This limit should be datasource-dependent.